src/deepsparse/image_classification/pipelines.py

```diff
@@ -0,0 +1,195 @@
+        """
+        Pre-Process the Inputs for DeepSparse Engine
+
+        :param inputs: input model
+        :return: list of preprocessed numpy arrays
+        """
+
+        image_batch = []
+
+        if isinstance(inputs.images, str):
+            inputs.images = [inputs.images]
+
+        for image in inputs.images:
+            img = cv2.imread(image) if isinstance(image, str) else image
+
+            img = cv2.resize(img, dsize=self.input_shape)
+            img = img[:, :, ::-1].transpose(2, 0, 1)
+
+            image_batch.append(img)
+
+        image_batch = numpy.stack(image_batch, axis=0)
+        image_batch = numpy.ascontiguousarray(image_batch, dtype=numpy.float32)
+        image_batch /= 255.0
+
+        # normalize entire batch
+        image_batch -= numpy.asarray(IMAGENET_RGB_MEANS).reshape((-1, 3, 1, 1))
+        image_batch /= numpy.asarray(IMAGENET_RGB_STDS).reshape((-1, 3, 1, 1))
+
+        return [image_batch]
+
+    def process_engine_outputs(
+        self,
+        engine_outputs: List[numpy.ndarray],
+    ) -> ImageClassificationOutput:
+        """
+        :param engine_outputs: list of numpy arrays that are the output of the engine
+            forward pass
+        :return: outputs of engine post-processed into an object in the `output_model`
+            format of this pipeline
+        """
+        labels = numpy.argmax(engine_outputs[0], axis=1).tolist()
+
+        if self.class_names is not None:
+            labels = [self.class_names[str(class_id)] for class_id in labels]
+
+        return ImageClassificationOutput(
+            scores=numpy.max(engine_outputs[0], axis=1).tolist(),
+            labels=labels,
+        )
+
+    @property
+    def input_model(self) -> Type[ImageClassificationInput]:
+        """
+        :return: pydantic model class that inputs to this pipeline must comply to
+        """
+        return ImageClassificationInput
+
+    @property
+    def output_model(self) -> Type[ImageClassificationOutput]:
+        """
+        :return: pydantic model class that outputs of this pipeline must comply to
+        """
+        return ImageClassificationOutput
+
+    @property
+    def input_shape(self) -> Tuple[int, ...]:
+        """
+        Returns the expected shape of the input tensor
+
+        :return: The expected shape of the input tensor from onnx graph
+        """
+        if self._input_shape is None:
+            self._input_shape = self._infer_input_shape()
+        return self._input_shape
+
+    def _infer_input_shape(self) -> Tuple[int, ...]:
+        """
+        Infer and return the expected shape of the input tensor
+
+        :return: The expected shape of the input tensor from onnx graph
+        """
+        onnx_model = onnx.load(self.engine.model_path)
+        input_tensor = onnx_model.graph.input[0]
+        return (
+            input_tensor.type.tensor_type.shape.dim[2].dim_value,
```

Comment:
Maybe I am completely misunderstanding this, but why are we talking only indices 2 and 3 here? 
I assume because 0 is batch size and 1 is no_channels right? I'd maybe incorporate no_channels into the inferred input shape. The reasoning would be -> because images sometimes tend to be 1D (some medical applications, segmentation).

Reply:
In that case, for clarity, I would change this function's name to `_infer_image_shape`. A bit of nit, but may clear some confusion for the user down the road.

Reply:
Thanks! changed!